url7f43a4752f_shalynnho_pokemonfactory_tgz-pJ8-Networking_ServerJ8/src/agent/KitProducerConsumerMonitor.java

```diff
@@ -0,0 +1,75 @@
+package agent;
+
+import java.util.Vector;
+
+import agent.data.Kit;
+
+
+/**
+ * Monitor that protects the shared Kits. Unused in v0.
+ * @author Daniel Paje
+ */
+public class KitProducerConsumerMonitor extends Object {
+	// Since there can only be 3 kits on the stand at any given time.
+	private final int N = 3;
+	private int count = 0;
+	private final Vector<Kit> theData;
+
+	synchronized public void insert(Kit data) {
+		while (count == 4) {
+			try {
+				System.out.println("\tFull, waiting");
+				wait(5000); // Full, wait to add
+			} catch (InterruptedException ex) {
+			}
+			;
+		}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 19:26, 39:45.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.